c/stcbackground.c

```diff
@@ -0,0 +1,73 @@
+#include "logging.h"
+#include "stcbackground.h"
+
+static int processStcBackgroundHandler(STCBase *base, STCModule *module, int selectStatus) {
+  BackgroundModuleData* moduleData = (BackgroundModuleData*)(module->data);
+  BackgroundTask* taskList = moduleData->taskList;
+
+  for (int i = 1; i < N_TASK_TABLE_ENTRIES; i++) {
+    BackgroundTask* t = &taskList[i];
+    if (t->id != 0)  /* initialise */ {
+      t->countInterval+=STC_BACKGROUND_INTERVAL;
+      if (t->countInterval == 0) {
+        Task task = t->task;
+        task(moduleData->server, t->taskInput);
+        t->countInterval=-t->timeInterval;
+      }
+    } else {
+      break;
+    }
+  }
+  return 0;
+};
+
+static int nextSlot(BackgroundTask* taskList) {
+  for (int i = 1; i < N_TASK_TABLE_ENTRIES; i++) {
```

Comment:
Why do we start from 1? Is it because `id = 0` indicates a uninitialized entry? Could we then just use -1 instead of 0?